### PR TITLE
feat: etc delete

### PR DESCRIPTION
**Pull request by FernandoAranda**
Issue #, if available:
GALI-1845
Description of changes:
Add soft delete for ETC

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-app/src/environmentTypeConfigRoutes.ts [hunk 1]
```diff
@@ -47,6 +54,48 @@ export function setUpEnvTypeConfigRoutes(
...
+        envTypeConfigId: req.params.envTypeConfigId
+      };
+      const validatedRequest = validateAndParse<DeleteEnvironmentTypeConfigRequest>(
+        DeleteEnvironmentTypeConfigRequestParser,
+        envTypeConfigDeleteRequest
+      );
+
+      async function checkDependency(envTypeId: string, envTypeConfigId: string): Promise<void> {
+        const typeId = `${resourceTypeToKey.envType}#${envTypeId}${resourceTypeToKey.envTypeConfig}#${envTypeConfigId}`;
+        const dependencies = await metadataService.listResourceByDependency<EnvironmentItem>(
```
Comment (sevu1) [inline solutions/swb-app/src/environmentTypeConfigRoutes.ts:72]: Do we need to list environments? I know it's more user friendly, unless @nguyen102 opposed to listing

File: workbench-core/environments/src/services/environmentTypeConfigService.ts [hunk 2]
```diff
@@ -28,6 +32,31 @@ export default class EnvironmentTypeConfigService {
...
+    await this.getEnvironmentTypeConfig(envTypeId, envTypeConfigId);
+
+    try {
+      await this._dynamoDbService.updateExecuteAndFormat({
+        key: this._buildEnvTypeConfigPkSk(envTypeConfigId),
+        params: {
+          item: { resourceType: `${this._resourceType}_deleted` }
+        }
+      });
+    } catch (e) {
```
Comment (sevu1) [inline workbench-core/environments/src/services/environmentTypeConfigService.ts:56]: We should log error. I wonder why there were no eslint warnings regarding unused variable

File: workbench-core/base/src/services/metadataService.ts [hunk 3]
```diff
@@ -58,6 +58,42 @@ export class MetadataService {
...
 
+  /**
+   *
+   * @param mainEntityResourceType - main entity resource type
+   * @param dependencyId - dependency entity id
+   * @param parser - Zod parser for metadata object
+   * @param queryParams - parameters to query metadata. Supported page size and pagination token.
+   * @returns object containind list of metadata objects and continuation token.
+   */
+  public async listResourceByDependency<DependencyMetadata extends { id: string }>(
```
[the remark this answers is not in the document]
Comment (FernandoAranda) [reply workbench-core/base/src/services/metadataService.ts]: we cannot as that one needs a pk to retrieve dependencies, we have the sk only , I deleted the method as I ddidn really needed it, I could use the environmentService for this

File: workbench-core/environments/src/services/environmentTypeConfigService.ts [hunk 4]
```diff
@@ -28,6 +32,31 @@ export default class EnvironmentTypeConfigService {
     this._dynamoDbService = dynamoDbService;
   }
 
+  /**
+   * Soft Environment Type Configuration
```
[the remark this answers is not in the document]
Comment (FernandoAranda) [reply workbench-core/environments/src/services/environmentTypeConfigService.ts]: fixed in last commit

**Review by sevu1 — APPROVED** [review]

**Review by DaveMadsen — APPROVED** [review]

**Pull request merged**